subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/GradleModuleMetadataParserTest.groovy

```diff
@@ -488,6 +488,33 @@ class GradleModuleMetadataParserTest extends Specification {
         0 * _
     }
 
+    def "parses requested capabilities without version"() {
+        def metadata = Mock(MutableModuleComponentResolveMetadata)
+        def variant = Mock(MutableComponentVariant)
+
+        when:
+        parser.parse(resource('''
+    {
+        "formatVersion": "1.1",
+        "variants": [
+            {
+                "attributes": { "usage": "runtime"},
+                "dependencies": [
+                    { "module": "m3", "group": "g3", "requestedCapabilities":[{"group":"org", "name":"foo"}]}
```

Comment:
Unless we are 100% positive that it is impossible there is metadata out there with a `"version: null"`, we should support parsing that as well.
Of course the parsing should be limited to that instance of a capability since it is not a possible case elsewhere.
makes sense?

Reply:
Thanks. This was indeed failing. I fixed it and extended this test case.